src/imports-parser.ts

```diff
@@ -0,0 +1,55 @@
+let rulesParser = require('./rules-parser');
+let fs = require('fs');
+/*
+  Imports file parser for split file systems
+  Note: Using a modified ES6 syntax to include imports
+*/
+export function parseWithImports(filename: string) {
+  // creating a stream through which each file will pass
+  let contents = fs.readFileSync(filename, "utf8");
+  return parserWrapper(contents, filename);
+}
+
+/*
+  *************************** Function Section ****************************
+*/
+
+/* ******** wrapper for recursive parsing ******* */
+function parserWrapper(data: string, filename: string) {
+  var sym = rulesParser.parse(data);
+
+  // Process any imports in symbol list
+  for (let i = 0; i < sym.imports.length; i++) {
+    let next = sym.imports[i];
+    var nextFilename = getNextFilenameFromContextAndImport(filename, next.filename);
+    let contents = fs.readFileSync(nextFilename, 'utf8');
+    let nextSymbols = parserWrapper(contents, nextFilename);
+    sym.imports[i].symbols = nextSymbols; // add it to the next part of the tree
+  }
+  return sym;
+}; // end function
+
+// Convert absolute filenames to relative
+// Convert relative filenames to include original path
+function getNextFilenameFromContextAndImport(current: string, nextImport: any) {
+  current = current.replace('.bolt', '');
+  nextImport = nextImport.replace('.bolt', '');
+  var currentFn = current.split('/');
+  var nextFn = nextImport.split('/');
+  let result = '';
+  if (nextFn[0] !== '.' && nextFn[0] !== '..') { // global reference
+    result = './node_modules/' + nextImport + '/index.bolt';
```

Comment:
Can we make the assumption that this is always ran from the root of the project directory?

Reply:
Hmm, interesting point. The difference here is between bower & npm (or npm dedupe). 

My preference would be not to have nesting (bower) but I didn't think of the nesting module issue. I can go back to the drawing board and look at the module resolution process to see if we can support an npm hierarchy module resultion.

Reply:
I don't think we need to support nesting, but maybe I'm not familiar enough is `./node_modules` going to exist? Is it possible to run this from a `./src/` directory so that we would need to look in `../node_modules`?

Reply:
node_modules are a standard npm spec so if they use NPM this would exist in the project root directory (not ./src/node_modules).

It's a valid test to load it from the ./src/myfile.bolt directory but generally the root directory is still considered ./

Reply:
I will have a look at using the following package to do the resolution https://www.npmjs.com/package/node-modules-resolve

Reply:
or this one is probably better https://www.npmjs.com/package/resolve

Reply:
Nice. It looks like that first package just recursively searches parent directories until it gets to a `./node_modules` and uses that one.

If the root directory is considered to be the current working directory we should check for it when we start. This makes an assumption which isn't enforced anywhere as far as I can tell.